invokeai/frontend/web/src/app/store/middleware/listenerMiddleware/listeners/imageMetadataReceived.ts

```diff
@@ -0,0 +1,26 @@
+import { invocationComplete } from 'services/events/actions';
+import { isImageOutput } from 'services/types/guards';
+import { imageMetadataReceived } from 'services/thunks/image';
+import { startAppListening } from '..';
+import { progressImageSet } from '../../../../../features/system/store/systemSlice';
+
+export const addImageMetadataReceivedListener = () => {
+  startAppListening({
+    predicate: (action) => {
+      if (
+        invocationComplete.match(action) &&
+        isImageOutput(action.payload.data.result)
+      ) {
+        return true;
+      }
+      return false;
+    },
+    effect: async (action, { getState, dispatch, take }) => {
+      if (imageMetadataReceived.fulfilled.match(action)) {
+        return;
+      }
+
+      dispatch(progressImageSet(null));
+    },
+  });
+};
```

Comment:
While this may fix the jumpiness, I don't think it's doing exactly what you want it to :P

The predicate must be true for the effect to run. In this case, the action caught by the middleware is the `invocationComplete` action, and there is an additional requirement that the result of the complete is an image.

So by the time you are in the effect, the action will never be a `imageMetadataReceived.fulfilled` action. So this will never return early - it will always `dispatch(progressImageSet(null))` when the predicate is true.